_script/test_couches.rb

```diff
@@ -0,0 +1,45 @@
+require "rubygems"
+require 'minitest/autorun'
+require './_script/verify_couches'
+require 'yaml'
+require 'nokogiri'
+
+class TestCouches < Minitest::Test
+    def setup
+        @couches = Couches.new
+        @country_codes = YAML.load_file("./_data/countries.yml")
+    end
+
+    def test_compulsory_fields
+        i = 0
+        @couches.all_front_matter.each do |front_matter|
+            assert((front_matter.class == Hash), "Incorrect format in front matter in #{@couches.couchnames[i]}")
+            assert(front_matter.has_key?("city"), "City field not found in #{@couches.couchnames[i]}")
+            assert(front_matter.has_key?("country"), "Country field not found in #{@couches.couchnames[i]}")
+            assert(front_matter.has_key?("email"), "Email field not found in #{@couches.couchnames[i]}")
+            i+=1
+        end 
+    end
+
+    def test_country_code
+        i = 0
+        @couches.all_front_matter.each do |front_matter|
+            if front_matter.has_key?("country")
+                code = (front_matter["country"])[/\s*[A-Z][A-Z]/] #Checking for 2 character country code in CAPITALS.
+                code_line = front_matter["country"]
+                assert(@country_codes.has_key?(code), "Invalid country code: '#{code_line}' found in #{@couches.couchnames[i]}.\nMake sure it is a valid 2 character country code in CAPS.")
+            end
+            i+=1
+        end
+    end
+
+    def test_content
+        i = 0
+        @couches.all_content.each do |content|
+            html_fragment = Nokogiri::HTML.fragment(content)
+            matches = html_fragment.search("h1, h2, h3, h4, h5, h6, p, li, ul") #Checking for HTML tags
```

Comment:
I'd want to run a sanitizer on top of this. https://github.com/flavorjones/loofah is a pretty good library that I've used in the past that handles this.

Things I don't want to let people do:

- Add their own styles
- Use h1 tags
- XSS things

Rest all should be fine for now, and we can look into things as people try them out. (`nav` for eg).